code/API_definitions/checkSimSwap-v0.3.0.camara.swagger.yaml

```
@@ -0,0 +1,288 @@
+security:
+  - oAuth2ClientCredentials: []
+  - BasicAuth: []
+  - apiKey: []
+  - three_legged:
+    - checkSimSwap
+    - getSimSwapDate
+servers:
+  - url: "{apiRoot}/{basePath}"
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root
+      basePath:
+        default: simSwap/v0
+        description: Base path for the simSwap API
+paths:
+  /retrieveSimSwapDate/:
+    post:
+      tags:
+        - Retrieve simSwap date
+      description: Get timestamp of last MSISDN <-> IMSI pairing change for a mobile user account provided with MSIDN.
+      operationId: createSimSwapDateRequest
+      requestBody:
+        description: |+
+          Create a simSwap date request. A MSISDN identifier is required.        
+        content :
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CreateSimSwapDate"
+        required: true
+      responses:
+        "200":
+          description: Contains information about sim swap change
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/SimSwapInfo"
+        "400":
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorInfo"
+              examples:
+                Generic400:
+                  summary: Schema validation failed
+                  value: 
+                    code: INVALID_INPUT
+                    message: "Schema validation failed at  ..."
+                MsisdnRequired:
+                  summary: msisdn is required as part of ueId
+                  value: 
+                    code: INVALID_INPUT
+                    message: "Expected property is missing: msisdn"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "409":
+          description: Conflict
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorInfo"
+              example:
+                code: CONFLICT
+                message: "Another resquest is created for the same UE"
+        "500":
+          description: Server error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorInfo"
+              example:
+                code: INTERNAL
+                message: "Session could not be created"
+        "503":
+          $ref: "#/components/responses/Generic503" 
+  /checkSimSwap:
+    post:
+      tags:
+      - Check simSwap
+      description: Manage request and response for a simSwap check for a mobile user account provided with MSIDN.
+      operationId: createCheckSimSwap
+      requestBody:
+        description: |+
+          Create a Check simSwap request. A MSISDN identifier and age are required.        

```

```suggestion
          Create a check SIM swap request for a MSISDN identifier.        
```

age discussed later but it could have a default to make API usage easier